cpp-netlib 0.12.0-rc1

theopolis opened the pull request:
### All Submissions:

- [X] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/homebrew/blob/master/.github/CONTRIBUTING.md) document?
- [X] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/homebrew/pulls) for the same update/change?

mikemcquaid commented on Library/Formula/cpp-netlib.rb:
```diff
@@ -1,9 +1,9 @@
-  version "0.11.2"
-  sha256 "71953379c5a6fab618cbda9ac6639d87b35cab0600a4450a7392bc08c930f2b1"
+  url "https://github.com/cpp-netlib/cpp-netlib/archive/cpp-netlib-0.12.0-rc1.tar.gz"
+  version "0.12.0-rc1"
```
If this is an `rc1` it should be in a `devel do` block instead.

theopolis replied:
Good call! I'm responsible for the only forward project dependency, but having RCs gated to devel makes a lot of sense given the brew devel documentation/guidance.

mikemcquaid commented on Library/Formula/cpp-netlib.rb:
```diff
@@ -15,6 +15,7 @@ class CppNetlib < Formula
 
   depends_on "cmake" => :build
   depends_on "openssl"
+  depends_on "asio"
```
Is this mandatory or optional?

theopolis replied:
Mandatory for this version 0.12.0, and most likely safe to assume 0.12.0+.

apjanke commented on Library/Formula/cpp-netlib.rb:
```diff
@@ -5,6 +5,12 @@ class CppNetlib < Formula
 
+  devel do
+    url "https://github.com/cpp-netlib/cpp-netlib/archive/cpp-netlib-0.12.0-rc1.tar.gz"
+    version "0.12.0"
```
Should this be `"0.12.0-rc1"`, to differentiate it from the eventual production release, and allow automatic `upgrade`s if an `rc2` comes out?

theopolis replied:
I agree, for some reason I thought the `-rc1` was dropping out from the `:version` define. But confirmed it does not.

apjanke commented on Library/Formula/cpp-netlib.rb:
```diff
@@ -27,7 +35,8 @@ class CppNetlib < Formula
     ENV.cxx11
 
-    system "cmake", *std_cmake_args
+    # NP: Do not build examples or tests as they require submodules.
```
Should this be "NB" ("nota bene") instead of "NP" ("nurse practitioner"/"no problem"/"non-polynomial")?

theopolis replied:
![image](https://cloud.githubusercontent.com/assets/981645/13620701/849f56c2-e546-11e5-9748-658f6a26ce9a.png)

apjanke commented:
This isn't something introduced by this change, but is `ENV.cflags.to_s.split` actually safe? IIRC, `$CFLAGS` could have quoted values with embedded spaces, and needs to be parsed as a substring of a shell command line, not a normal split-on-whitespace string.

theopolis commented:
Seems like it's "not that prevalent" within the formula base:
```
./Library/Aliases/libgme:43:                   "-lgme", "-o", "test", *ENV.cflags.to_s.split
./Library/Aliases/sqlite3:76:                     *ENV.cflags.to_s.split
./Library/Formula/cpp-netlib.rb:70:    ] + ENV.cflags.to_s.split
./Library/Formula/cpprestsdk.rb:40:             "-lboost_filesystem", "-lcpprest"] + ENV.cflags.to_s.split
./Library/Formula/fftw.rb:83:    system ENV.cc, "-o", "fftw", "fftw.c", "-lfftw3", *ENV.cflags.to_s.split
./Library/Formula/game-music-emu.rb:43:                   "-lgme", "-o", "test", *ENV.cflags.to_s.split
./Library/Formula/glbinding.rb:33:                    "-lglbinding", *ENV.cflags.to_s.split
./Library/Formula/glib.rb:142:    system ENV.cc, "-o", "test", "test.c", *(flags + ENV.cflags.to_s.split)
./Library/Formula/google-benchmark.rb:40:    flags = ["-stdlib=libc++", "-I#{include}", "-L#{lib}", "-lbenchmark"] + ENV.cflags.to_s.split
./Library/Formula/hidapi.rb:47:    flags = ["-I#{include}/hidapi", "-L#{lib}", "-lhidapi"] + ENV.cflags.to_s.split
./Library/Formula/leptonica.rb:55:    flags = ["-I#{include}/leptonica"] + ENV.cflags.to_s.split
./Library/Formula/libcapn.rb:42:    flags = ["-I#{include}/capn", "-L#{lib}/capn", "-lcapn"] + ENV.cflags.to_s.split
./Library/Formula/libffi.rb:86:    system ENV.cc, "-o", "closure", "closure.c", *(flags + ENV.cflags.to_s.split)
./Library/Formula/liboil.rb:41:    flags = ["-I#{include}/liboil-0.3", "-L#{lib}", "-loil-0.3"] + ENV.cflags.to_s.split
./Library/Formula/libstrophe.rb:55:    system ENV.cc, "-o", "test", "test.c", *(flags + ENV.cflags.to_s.split)
./Library/Formula/sqlite.rb:76:                     *ENV.cflags.to_s.split
./Library/Formula/torrentcheck.rb:9:    system ENV.cc, "torrentcheck.c", "sha1.c", "-o", "torrentcheck", *ENV.cflags.to_s.split
```

But it seems like the right thing is to drop the string flattening.

apjanke commented:
I dunno if `ENV.cflags.split` is right either – I think `ENV.cflags` is already a string, so it seems like the `split` which may be problematic, since it won't respect embedded quotes. 

Is it necessary to pass `ENV.cflags` as arguments to `cxx`, if they're already in the environment variable?

theopolis commented:
If they aren't already in the environment, then I'd assume it is-- provided a compile flag is needed for the tests to compile. ;) But if they 'are' in the environment then it depends on whether the environment is preserved when the fork to run the compiler is called.

In this case the additional flags from the brew ENV container are not needed for the test. But your question is good and the approach here mimics a ton of other formulas. If this is a concern, perhaps a codemod is in your future.

Is the observation here a blocker for merging?

apjanke commented:
> Is the observation here a blocker for merging?

Nope. Wasn't introduced by this change and isn't breaking anything; no reason to wait on it.

DomT4 commented on Library/Formula/cpp-netlib.rb:
```diff
@@ -15,6 +21,8 @@ class CppNetlib < Formula
   depends_on "cmake" => :build
   depends_on "openssl"
+  # Version 0.12.0+ move from boost::asio to asio.
+  depends_on "asio"  if build.devel?
```
This should be in the `devel do` block FYI, not tagged with `if build.devel?` here.

apjanke commented:
Merged. Thank you for your contribution to Homebrew!
